be/src/vec/exec/scan/scanner_scheduler.cpp

```diff
@@ -289,6 +297,7 @@ void ScannerScheduler::_scanner_scan(std::shared_ptr<ScannerContext> ctx,
                     scan_task->cached_blocks.back().first.get()->set_columns(
                             std::move(mutable_block.mutable_columns()));
                     ctx->return_free_block(std::move(free_block));
+                    ctx->update_max_memory_usage_at_same_time(-free_block_bytes);
                     ctx->inc_free_block_usage(
                             scan_task->cached_blocks.back().first->allocated_bytes() - block_size);
```

Comment:
scan_task->cached_blocks.back().first->allocated_bytes() - block_size); 
why not using freee block bytes directly???